cloudevents/pydantic/event.py

```diff
@@ -0,0 +1,259 @@
+#  Copyright 2018-Present The CloudEvents Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import datetime
+import typing
+
+import pydantic
+
+import cloudevents.conversion
+from cloudevents import abstract, conversion
+from cloudevents.sdk.event import attribute
+
+
+def _ce_json_dumps(obj: typing.Dict[str, typing.Any], *args, **kwargs) -> str:  # noqa
+    return conversion.to_json(
+        cloudevents.conversion.from_dict(
+            CloudEvent,
+            obj,
+        )
+    ).decode("utf-8")
+
+
+def _ce_json_loads(
+    data: typing.Union[str, bytes], *args, **kwargs  # noqa
+) -> typing.Dict[typing.Any, typing.Any]:
+    return cloudevents.conversion.to_dict(conversion.from_json(CloudEvent, data))
+
+
+class CloudEvent(abstract.CloudEvent, pydantic.BaseModel):
+
+    """
+    Python-friendly cloudevent class supporting v1 events
+    Supports both binary and structured mode CloudEvents
+    """
+
+    @classmethod
+    def create(
+        cls, attributes: typing.Dict[str, typing.Any], data: typing.Optional[typing.Any]
+    ) -> "CloudEvent":
+        return cls(attributes, data)
+
+    data: typing.Optional[typing.Any] = pydantic.Field(
```

Comment:
Can you please add `examples` attribute to the `pydantic.Field` here and for other fields?

That's super useful when rendering OpenAPI or JSON schemas from the model.